### PR TITLE
Miscellaneous bug fixes

### PR DESCRIPTION
**Pull request by GlowingUmbreon**
This pull request fixes various bugs in different parts of the program. Looking at the commits should give more information for each fix.

### CODE REVIEW

File: .env.sample [hunk 1]
```diff
@@ -1,44 +1,56 @@
...
 SERVER_COOKIE_SECRET=
 
+# Encryption
 ENCRYPTION_KEY=
 CIPHERSWEET_KEY=
 
+# Admin panel
 TOOTSIE_PASSWORD=
 
+# MariaDB
```
Comment (lightbulblighter) [inline .env.sample:26]: The average sysadmin doesn't really need to know what the internal database system is. You can make this just say "Database."

File: .env.sample [hunk 2]
```diff
@@ -1,44 +1,56 @@
...
 
+# Admin panel
 TOOTSIE_PASSWORD=
 
+# MariaDB 
 DB_PASSWORD=
 DB_USERNAME=root
 DB_NAME=rboxlo
 
+# Access permissions
```
Comment (lightbulblighter) [inline .env.sample:31]: Should just be "visibility."

File: .env.sample [hunk 3]
```diff
@@ -1,44 +1,56 @@
...
 PRIVATE_REGISTRATION=true
 PRIVATE_REFERRAL=false
 PRIVATE_INVITE_ONLY=false
 PRIVATE_CLOSED=false
 
+# Email
 MAIL_PASSWORD=
 MAIL_USERNAME=no-reply
 
+# Captcha
```
Comment (lightbulblighter) [inline .env.sample:42]: This is all Google related APIs (see the analytics) and should be labeled properly.

File: Server/lib/base/locker.js [hunk 4]
```diff
@@ -8,6 +8,9 @@ if (global.rboxlo.env.ENCRYPTION_KEY.length != 32) {
     // Will result in buffer exceptions if not EXACTLY thirty-two characters
     throw `Invalid env.ENCRYPTION_KEY length (expected 32, got ${global.rboxlo.env.ENCRYPTION_KEY.length})`
 }
+if (global.rboxlo.env.CIPHERSWEET_KEY.length != 64) {
```
Comment (lightbulblighter) [inline Server/lib/base/locker.js:11]: There should be a newline before the previous check. Also, length isn't everything when it comes to the ciphersweet key. It should be a valid hex string and a Buffer should be able to be created out of the key. A try catch would possibly work, or some other validation check to see if it's a hex string.

**Review by lightbulblighter — CHANGES_REQUESTED** [review]
Some minor stuff that needs changes. Once fixed will be merged.

File: Server/lib/base/locker.js (hunk 4 above)
Comment (GlowingUmbreon) [reply Server/lib/base/locker.js:11]: I know but its just a quick check to make sure that the user actually fills it in.

File: .env.sample (hunk 2 above)
Comment (GlowingUmbreon) [reply .env.sample:31]: Also I noticed that PRIVATE_REGISTRATION is the wrong way round, if its true it allows registration but if its false it does not.

File: Server/lib/base/locker.js (hunk 4 above)
Comment (lightbulblighter) [reply Server/lib/base/locker.js:11]: Sounds good.

**Comment by GlowingUmbreon** [comment]
Okay the changes requested should be in the pull request now.

**Pull request merged**